site/src/api/queries/workspaces.ts

```diff
@@ -18,13 +21,11 @@ export const workspaceByOwnerAndNameKey = (owner: string, name: string) => [
   "settings",
 ];
 
-export const workspaceByOwnerAndName = (
-  owner: string,
-  name: string,
-): QueryOptions<Workspace> => {
```

Comment:
It was because when using the `queryKey` value from the `QueryOptions` it was returning a type error because it can be undefined:

Example:
```ts
queryClient.setQueryData(
  workspaceQueryOptions.queryKey,
  newWorkspaceData,
);
```

Error:
```console
Argument of type 'QueryKey | undefined' is not assignable to parameter of type 'QueryKey'
```

I could try to better type it but for this case, I just think it is too much. At some point, we may create a utility type to help us with typing the query results.

Reply:
Yeah, RQ v5's new `queryOptions` function might also help with this